### PR TITLE
Give section editors access to list views

### LINKED ISSUE #2229: Search Submissions for Section Editors
Allow section editors to see and use the Search Submissions screen.

### PR DESCRIPTION
**Pull request by joemull**
Closes #2229.

I talked to three editors who had worked with section or guest editors, and the consensus was that they should be able to search submissions and view the workflow list pages for review and copyediting, but only the articles to which they are assigned should be visible.

### CODE REVIEW

File: src/core/logic.py [hunk 1]
```diff
@@ -919,3 +919,11 @@ def render_nested_setting(
     )
 
     return rendered_string
+
```
Comment (ajrbyers) [inline src/core/logic.py:922]: There should be an extra newline here.

File: src/core/logic.py [hunk 2]
```diff
@@ -919,3 +919,11 @@ def render_nested_setting(
...
+def restrict_articles_to_editor_assigned(request, articles):
```
Comment (ajrbyers) [inline src/core/logic.py:923]: Might be worth renaming this, it doesn't actually restrict anything and could be used for other things.

File: src/core/templatetags/nose_tools.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
+from nose import tools
+from django import template
+
+register = template.Library()
+
+class NoseNode(template.Node):
+
+    def render(self, context):
+        tools.set_trace()         # Debugger will stop here
```
Comment (ajrbyers) [inline src/core/templatetags/nose_tools.py:9]: This is nice!

**Review by ajrbyers — APPROVED** [review]

**Comment by ajrbyers** [comment]
This looks good, a few comments and a conflict to resolve.

**Comment by joemull** [comment]
After it was approved, this branch was corrupted with a rebase that did not do what we wanted. It is superseded by https://github.com/BirkbeckCTP/janeway/pull/3115

**Pull request closed without merging**